src/Raven.Server/Documents/ETL/FilterTimeSeriesDeleteRangeEnumerator.cs

```diff
@@ -0,0 +1,96 @@
+using System.Collections;
+using System.Collections.Generic;
+using Raven.Server.Documents.ETL.Stats;
+using Raven.Server.Documents.Replication.ReplicationItems;
+using Raven.Server.Documents.TimeSeries;
+using Raven.Server.ServerWide.Context;
+
+namespace Raven.Server.Documents.ETL
+{
+    //TODO to make base class for filter time-series (and maybe counters too)
+    public class FilterTimeSeriesDeleteRangeEnumerator : IEnumerator<TimeSeriesDeletedRangeItem>
+    {
+        private readonly IEnumerator<TimeSeriesDeletedRangeItem> _timeSeries;
+        private readonly EtlStatsScope _stats;
+        private readonly DocumentsStorage _docsStorage;
+        private readonly DocumentsOperationContext _context;
+        private readonly long _lastProcessedDocEtagInBatch;
+
+        public FilterTimeSeriesDeleteRangeEnumerator(IEnumerator<TimeSeriesDeletedRangeItem> timeSeries, EtlStatsScope stats, DocumentsStorage docsStorage, DocumentsOperationContext context, long lastProcessedDocEtagInBatch)
+        {
+            //TODO To check if should filter on collection
+            _timeSeries = timeSeries;
+            _stats = stats;
+            _docsStorage = docsStorage;
+            _context = context;
+            _lastProcessedDocEtagInBatch = lastProcessedDocEtagInBatch;
+        }
+
+        public bool MoveNext()
+        {
+            Current = null;
+
+            while (_timeSeries.MoveNext())
+            {
+                var current = _timeSeries.Current;
+
+                if (CanMoveNextAndNotExceedLastDocumentInBatch(current.Etag, _lastProcessedDocEtagInBatch) == false)
+                    return false;
+
+                TimeSeriesValuesSegment.ParseTimeSeriesKey(current.Key, _context, out var docId, out var name);
+                var doc = _docsStorage.Get(_context, docId);
+                if (doc != null && current.Etag > doc.Etag)
```

Comment:
Why do you have this check?
A time series can appear both before and after the document etag.